From acb30582e3f9b065fdc01dd26bdb49579994a9c1 Mon Sep 17 00:00:00 2001
From: Romain Naour <romain.naour@openwide.fr>
Date: Sat, 11 Apr 2015 12:46:05 +0200
Subject: [PATCH 1/2] autotools: add --disable-libeeze

When cross-compiling, we only want to build edje_cc, embryo_cc
and eet binaries for the host before starting the build for the
target.

This patch allows to disable libeeze in order to shorten the
build time but most of all remove the dependency on libudev.

In normal case it's not recommended hence a warning.

Signed-off-by: Romain Naour <romain.naour@openwide.fr>
Cc: Cedric BAIL <cedric@osg.samsung.com>
---
 configure.ac | 34 +++++++++++++++++++++++++++++++++-
 1 file changed, 33 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index a0dc51e..c5ead11 100644
--- a/configure.ac
+++ b/configure.ac
@@ -617,6 +617,31 @@ AC_DEFINE_IF([HAVE_ATFILE_SOURCE],
 
 ######################  EFL  ######################
 
+want_libeeze="yes"
+AC_ARG_ENABLE([libeeze],
+   [AS_HELP_STRING([--disable-libeeze],[disable Eeze device library @<:@default=enable@:>@])],
+   [
+    if test "x${enableval}" = "xyes"; then
+       want_libeeze="yes"
+    else
+       want_libeeze="no"
+       CFOPT_WARNING="yes"
+    fi
+   ])
+
+build_libeeze="yes"
+if test "x${want_libeeze}" = "xyes" ; then
+    if test "x${have_linux}" = "xyes" ; then
+        build_libeeze="yes"
+    else
+        build_libeeze="no"
+        want_libeeze="no"
+        CFOPT_WARNING="yes"
+    fi
+else
+    build_libeeze="no"
+fi
+
 AC_ARG_ENABLE([systemd],
    [AS_HELP_STRING([--enable-systemd],[Enable systemd support. @<:@default=disabled@:>@])],
    [
@@ -2965,7 +2990,7 @@ have_libmount_new="no"
 have_libmount_old="no"
 have_eeze_mount="no"
 
-EFL_LIB_START_OPTIONAL([Eeze], [test "${have_linux}" = "yes"])
+EFL_LIB_START_OPTIONAL([Eeze], [test "x${build_libeeze}" = "xyes"])
 
 ### Additional options to configure
 AC_ARG_WITH([mount],
@@ -4993,6 +5018,13 @@ if test -n "$CFOPT_WARNING"; then
     echo "to both as it will break your build if set to eo."
     echo "_____________________________________________________________________"
   fi
+  if test "x${want_libeeze}" = "xno"; then
+    echo "_____________________________________________________________________"
+    echo "Libeeze has been disabled, and it is used heavily for support of"
+    echo "removable devices etc. and disabling this will hurt support for"
+    echo "Enlightenment and its filemanager."
+    echo "_____________________________________________________________________"
+  fi
   echo "_____________________________________________________________________"
   echo ""
   echo "==-- WARNING --=="
-- 
1.9.3

